File: Csharp/Show/AbstractFactory/ProductB1.cs
namespace AbstractFactory
{
    public class ProductB1: ProductB
    {
        public string UsefulFunctionB()
        {
            return "The result of the product B1.";
        }

        public string AnotherUsefulFunctionB(ProductA collaborator)
        {
            var result = collaborator.UsefulFunctionA();
            return $"The result of the B1 collaborating with the ({result})";
        }
    }
}

File: JavaScript/Creational/AbstractFactory/Example1/ProductB1.js
const AbstractProductB = require('./AbstractProductB');

class ProductB1 extends AbstractProductB {
    usefulFunctionB() {
        return 'ProductB1: The result of the product B1.';
    }
    anotherUsefulFunctionB(collaborator) {
        return `ProductB1: Collaborating with (${collaborator.usefulFunctionA()})`;
    }
}
module.exports = ProductB1;